robot/impl/local_robot.go

```diff
@@ -1039,20 +1037,6 @@ func (r *localRobot) Reconfigure(ctx context.Context, newConfig *config.Config)
 		}
 	}
 
-	// Sync Packages before reconfiguring rest of robot and resolving references to any packages
-	// in the config.
-	// TODO(RSDK-1849): Make this non-blocking so other resources that do not require packages can run before package sync finishes.
-	// TODO(RSDK-2710) this should really use Reconfigure for the package and should allow itself to check
-	// if anything has changed.
-	err := r.packageManager.Sync(ctx, newConfig.Packages)
-	if err != nil {
-		allErrs = multierr.Combine(allErrs, err)
-	}
-
-	err = r.replacePackageReferencesWithPaths(newConfig)
```

Comment:
[q] why are you removing this line here? can't see it being added where we added sync.

Reply:
Added to config/reader.go